gateway/ln-gateway/src/actor.rs

```diff
@@ -147,153 +143,204 @@ impl GatewayActor {
         }
     }
 
-    pub async fn subscribe_htlcs(&mut self) -> Result<()> {
+    async fn subscribe_to_htlcs(
+        &self,
+        ln_sender: Sender<RouteHtlcRequest>,
+        short_channel_id: u64,
+    ) -> Result<()> {
```

Comment:
This fn feels like it should be called on `ln_sender` as `self` and return a stream of intercepted HTLCs. But it was like this already before, so not this PRs concern.

Reply:
This was easy enough to do. I moved `settle_htlc` and `cancel_htlc` into the new struct too.